src/TicketContext.tsx

```diff
@@ -56,6 +69,19 @@ const ticketReducer: TicketReducer = (prevState, action) => {
         isRefreshingTickets: false,
       };
     }
+    case 'SET_IS_REFRESHING_PREASSIGNED_FARE_PRODUCTS': {
+      return {
+        ...prevState,
+        isRefreshingPreassignedFareProducts: true,
+      };
+    }
```

Comment:
Do you need `SET_IS_REFRESHING_PREASSIGNED_FARE_PRODUCTS` as its own action type? Since it only happens on startup, couldn't `isRefreshingPreassignedFareProducts` be set as `true` in `initialReducerState` further down?